src/libraries/Microsoft.PowerFx.Core/Functions/FunctionScopeInfo.cs

```diff
@@ -200,6 +215,26 @@ public void CheckLiteralPredicates(TexlNode[] args, IErrorContainer errors)
                     }
                 }
             }
+        }
+
+        /// <summary>
+        /// Get the scope identifiers for the function based on the CallNode child args.
+        /// The majority of the functions will have a single scope identifier named ThisRecord.
+        /// Other functions, like Join, may create 2 or more scope identifiers.
+        /// </summary>
+        /// <param name="nodes">Call child nodes.</param>
+        /// <param name="scopeIdents">Scope names.</param>
+        /// <returns></returns>
+        public virtual bool GetScopeIdent(TexlNode[] nodes, out DName[] scopeIdents)
+        {
+            scopeIdents = new[] { TexlBinding.ThisRecordDefaultName };
+            if (nodes[0] is AsNode asNode)
+            {
+                scopeIdents = new[] { asNode.Right.Name };
+                return true;
+            }
+
+            return false;
         }
```

Comment:
When does this return true/false, how is that used? should this always return scopeIdents?

Reply:
It always return scopeIdents. The return will dictate if binder will work with "standard" scope names (ThisRecord, LeftRecord, RightRecord) or a alias (`t1 As NewAlias`).